caps/types.go

```diff
@@ -65,3 +63,18 @@ func (t *Type) Validate(c *Capability) error {
 	}
 	return nil
 }
+
+// TypeLookupFn aids in looking up a Type by name
+type TypeLookupFn func(name string) *Type
+
+// MarshalJSON encodes a Type object as the name of the type.
+func (t *Type) MarshalJSON() ([]byte, error) {
+	return json.Marshal(t.Name)
+}
+
+// UnmarshalJSON decodes the name of a Type object.
+// NOTE: In the future, when more properties are added, those properties will
+// not be decoded and will be left over as empty values.
+func (t *Type) UnmarshalJSON(data []byte) error {
+	return json.Unmarshal(data, &t.Name)
```

Comment:
Hmm.. seems okay for now, but it's a bit eye-catching. It's loading a broken type. I think we'll need to return to this later.